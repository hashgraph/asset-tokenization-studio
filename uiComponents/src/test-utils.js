import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
