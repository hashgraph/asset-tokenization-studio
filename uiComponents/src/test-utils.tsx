import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
