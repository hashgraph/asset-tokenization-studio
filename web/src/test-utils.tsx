import React from "react";
import { ChakraProvider } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render, waitFor } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import i18n from "./i18n";
import theme from "./theme";
import userEvent from "@testing-library/user-event";

export const selectCalendar = async (
  component: RenderResult,
  id: string,
  day: string | number = new Date().getDate(),
) => {
  const calendar = component.getByTestId(id);
  await userEvent.click(calendar);

  await waitFor(() => {
    const daysToSelect = component.getAllByTestId(`day-${day}`);

    const dayToSelect = daysToSelect.find(
      (day) => !day.hasAttribute("disabled"),
    );

    if (dayToSelect) {
      userEvent.click(dayToSelect);
    }
  });

  await userEvent.click(document.body);
};

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: () => {},
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory,
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <I18nextProvider i18n={i18n}>
      <QueryClientProvider client={testQueryClient}>
        <ChakraProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            {children}
          </Router>
        </ChakraProvider>
      </QueryClientProvider>
    </I18nextProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history,
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {},
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllProviders history={history}>{children}</AllProviders>
    ),
    ...options,
  });

export { customRender as render };
