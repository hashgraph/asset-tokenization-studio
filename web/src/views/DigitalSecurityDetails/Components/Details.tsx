import { Box, Center, HStack, VStack } from "@chakra-ui/react";
import { Text } from "@iob/io-bricks-ui/Foundations";
import { DefinitionList } from "@iob/io-bricks-ui/DataDisplay";
import { Panel } from "../../../components/Panel";
import { useTranslation } from "react-i18next";
import { ActionsButtons } from "./ActionsButtons";
import {
  BondDetailsViewModel,
  EquityDetailsViewModel,
  GetAccountBalanceRequest,
  GetAllCouponsRequest,
  GetAllDividendsRequest,
  SecurityViewModel,
} from "@iob/securitytoken-sdk";
import { formatNumberLocale, toNumber } from "../../../utils/format";
import { useUserStore } from "../../../store/userStore";
import { COUPONS_FACTOR, User } from "../../../utils/constants";
import { useSecurityStore } from "../../../store/securityStore";
import { SkeletonText } from "@chakra-ui/react";
import {
  useGetAllDividends,
  useGetBalanceOf,
} from "../../../hooks/queries/useGetSecurityDetails";
import { useWalletStore } from "../../../store/walletStore";
import { SecurityDetailsExtended } from "./SecurityDetailsExtended";
import { useGetAllCoupons } from "../../../hooks/queries/useCoupons";
import { CountriesList } from "../../CreateSecurityCommons/CountriesList";
import _capitalize from "lodash/capitalize";

interface DetailsProps {
  id?: string;
  detailsResponse: SecurityViewModel;
  equityDetailsResponse: EquityDetailsViewModel;
  bondDetailsResponse: BondDetailsViewModel;
}

export const Details = ({
  id = "",
  equityDetailsResponse,
  bondDetailsResponse,
}: DetailsProps) => {
  const { t: tProperties } = useTranslation("properties");
  const { t: tPermissions } = useTranslation("properties", {
    keyPrefix: "permissions",
  });
  const { t: tRights } = useTranslation("security", {
    keyPrefix: "createEquity",
  });
  const { t: tBenefits } = useTranslation("security", {
    keyPrefix: "details.benefits",
  });
  const { t: tRegulations } = useTranslation("properties", {
    keyPrefix: "regulations",
  });
  const { t: tRegulation } = useTranslation("security", {
    keyPrefix: "regulation",
  });

  const { type: userType } = useUserStore();
  const { details = {} } = useSecurityStore();
  const { address: walletAddress } = useWalletStore();

  // DIVIDENDS
  const dividendsRequest = new GetAllDividendsRequest({
    securityId: id,
  });
  const { data: dividends } = useGetAllDividends(dividendsRequest);

  // COUPONS
  const couponsRequest = new GetAllCouponsRequest({
    securityId: id,
  });
  const { data: coupons } = useGetAllCoupons(couponsRequest);

  const rightsAndPrivileges = {
    votingRights: equityDetailsResponse.votingRight,
    informationRights: equityDetailsResponse.informationRight,
    liquidationRights: equityDetailsResponse.liquidationRight,
    subscriptionRights: equityDetailsResponse.subscriptionRight,
    conversionRights: equityDetailsResponse.convertionRight,
    redemptionRights: equityDetailsResponse.redemptionRight,
    putRights: equityDetailsResponse.putRight,
  };

  const rightsAndPrivilegesFiltered = Object.entries(rightsAndPrivileges)
    .filter(([_key, value]) => value)
    .map(([key]) => tRights(`stepNewSerie.${key}`));

  const balanceOfRequest = new GetAccountBalanceRequest({
    securityId: id,
    targetId: walletAddress,
  });

  const { data: availableBalance, isLoading: isAvailableBalanceLoading } =
    useGetBalanceOf(balanceOfRequest);

  const permissionsItems = [
    {
      title: tPermissions("controllable"),
      description: details?.isControllable
        ? tPermissions("allowed")
        : tPermissions("notAllowed"),
    },
    {
      title: tPermissions("blocklist"),
      description: details?.isWhiteList
        ? tPermissions("notAllowed")
        : tPermissions("allowed"),
    },
    {
      title: tPermissions("approvalList"),
      description: details?.isWhiteList
        ? tPermissions("allowed")
        : tPermissions("notAllowed"),
    },
  ];

  const {
    type,
    subType = "",
    dealSize = "",
    accreditedInvestors = "",
    maxNonAccreditedInvestors = "",
    manualInvestorVerification = "",
    internationalInvestors = "",
    resaleHoldPeriod = "",
  } = details?.regulation || {};

  const regulationItems = [
    {
      title: tRegulations("regulationType"),
      description: tRegulation(`regulationType_${type}`),
    },
    {
      title: tRegulations("regulationSubType"),
      description: subType.replace("_", " "),
    },
    {
      title: details?.isCountryControlListWhiteList
        ? tRegulations("allowedCountries")
        : tRegulations("blockedCountries"),
      description:
        details?.countries
          ?.split(",")
          .map(
            (country) => CountriesList[country as keyof typeof CountriesList],
          )
          .join(" - ") ?? "",
    },
    {
      title: tRegulations("dealSize"),
      description:
        dealSize !== "0"
          ? `${dealSize} $`
          : tRegulations("dealSizePlaceHolder"),
    },
    {
      title: tRegulations("accreditedInvestors"),
      description: _capitalize(accreditedInvestors),
    },
    {
      title: tRegulations("maxNonAccreditedInvestors"),
      description:
        maxNonAccreditedInvestors !== 0
          ? `${maxNonAccreditedInvestors}`
          : tRegulations("maxNonAccreditedInvestorsPlaceHolder"),
    },
    {
      title: tRegulations("manualInvestorVerification"),
      description: _capitalize(manualInvestorVerification),
    },
    {
      title: tRegulations("internationalInvestors"),
      description: _capitalize(internationalInvestors),
    },
    {
      title: tRegulations("resaleHoldPeriod"),
      description: _capitalize(resaleHoldPeriod),
    },
  ];

  if (Object.keys(equityDetailsResponse).length > 0) {
    permissionsItems.push({
      title: tPermissions("rightsAndPrivileges"),
      description: rightsAndPrivilegesFiltered.join(", "),
    });
  }

  const currentAvailableBalance = toNumber(availableBalance?.value);

  const nominalValue = toNumber(
    equityDetailsResponse?.nominalValue || bondDetailsResponse?.nominalValue,
    2,
  );

  const totalAvailableBalanceValue = nominalValue * currentAvailableBalance;

  const nominalTotalValue = formatNumberLocale(
    nominalValue * ((details?.totalSupply ?? 0) as number),
    2,
  );

  return (
    <VStack gap={6} pt={2} pb={8} w="full">
      <ActionsButtons />
      <HStack w="full" gap={8} alignItems="flex-start">
        <VStack w="full" gap={8}>
          <SecurityDetailsExtended
            layerStyle="container"
            nominalValue={nominalValue}
          />
          {dividends?.map((dividend) => (
            <DefinitionList
              key={dividend.dividendId}
              items={[
                {
                  title: tBenefits("id"),
                  description: dividend.dividendId ?? "",
                },
                {
                  title: tBenefits("recordDate"),
                  description:
                    new Date(
                      dividend.recordDate.getTime(),
                    ).toLocaleDateString() ?? "",
                },
                {
                  title: tBenefits("executionDate"),
                  description:
                    new Date(
                      dividend.executionDate.getTime(),
                    ).toLocaleDateString() ?? "",
                },
                {
                  title: tBenefits("dividendAmount"),
                  description: dividend.amountPerUnitOfSecurity ?? "",
                },
                ...(dividend.snapshotId
                  ? [
                      {
                        title: "Snapshot Id",
                        description: dividend.snapshotId,
                      },
                    ]
                  : []),
              ]}
              title={tBenefits("dividends")}
              layerStyle="container"
            />
          ))}
          {coupons?.map((coupon) => (
            <DefinitionList
              key={coupon.couponId}
              items={[
                {
                  title: tBenefits("id"),
                  description: coupon.couponId ?? "",
                },
                {
                  title: tBenefits("recordDate"),
                  description:
                    new Date(
                      coupon.recordDate.getTime(),
                    ).toLocaleDateString() ?? "",
                },
                {
                  title: tBenefits("executionDate"),
                  description:
                    new Date(
                      coupon.executionDate.getTime(),
                    ).toLocaleDateString() ?? "",
                },
                {
                  title: tBenefits("couponRate"),
                  description:
                    `${parseInt(coupon.rate) / COUPONS_FACTOR}%` ?? "",
                },
                ...(coupon.snapshotId
                  ? [
                      {
                        title: tBenefits("snapshot"),
                        description: coupon.snapshotId,
                      },
                    ]
                  : []),
              ]}
              title={tBenefits("coupons")}
              layerStyle="container"
            />
          ))}
        </VStack>
        <VStack w="full" gap={8}>
          {userType === User.holder && (
            <Panel title={tProperties("currentAvailableBalance")}>
              <Center w="full">
                {isAvailableBalanceLoading ? (
                  <SkeletonText skeletonHeight={2} flex={1} noOfLines={1} />
                ) : (
                  <Box>
                    <Text textStyle="ElementsSemibold2XL">
                      {currentAvailableBalance}
                      <Text ml={1} as="span" textStyle="ElementsRegularMD">
                        {details?.symbol ?? ""}
                      </Text>
                    </Text>
                    <Text ml={1} as="span" textStyle="ElementsRegularMD">
                      {tProperties("nominalTotalValue")}
                      {formatNumberLocale(totalAvailableBalanceValue, 2)}
                    </Text>
                  </Box>
                )}
              </Center>
            </Panel>
          )}
          <Panel title={tProperties("totalSupply")}>
            <Center w="full">
              {details ? (
                <Box>
                  <Text textStyle="ElementsSemibold2XL">
                    {details?.totalSupply ?? ""}
                    <Text ml={1} as="span" textStyle="ElementsRegularMD">
                      {details?.symbol ?? ""}
                    </Text>
                  </Text>
                  <Text ml={1} as="span" textStyle="ElementsRegularMD">
                    {tProperties("nominalTotalValue")}
                    {nominalTotalValue}
                  </Text>
                </Box>
              ) : (
                <SkeletonText skeletonHeight={2} flex={1} noOfLines={1} />
              )}
            </Center>
          </Panel>
          <DefinitionList
            isLoading={details === null}
            items={permissionsItems}
            title={tPermissions("label")}
            layerStyle="container"
          />

          <DefinitionList
            isLoading={details === null}
            items={regulationItems}
            title={tRegulations("label")}
            layerStyle="container"
          />
        </VStack>
      </HStack>
    </VStack>
  );
};
