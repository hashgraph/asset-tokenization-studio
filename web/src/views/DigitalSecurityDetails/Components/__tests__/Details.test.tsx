import { Details } from "../Details";
import { render } from "../../../../test-utils";
import {
  BondDetailsViewModel,
  EquityDetailsViewModel,
  SecurityViewModel,
} from "@iob/securitytoken-sdk";

// TODO Improve tests when it is connected to SDK
describe(`${Details.name}`, () => {
  test("should render correctly", () => {
    const detailsResponse: SecurityViewModel = {
      name: "test",
    };

    const component = render(
      <Details
        id=""
        detailsResponse={detailsResponse}
        equityDetailsResponse={{} as EquityDetailsViewModel}
        bondDetailsResponse={{} as BondDetailsViewModel}
      />,
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
