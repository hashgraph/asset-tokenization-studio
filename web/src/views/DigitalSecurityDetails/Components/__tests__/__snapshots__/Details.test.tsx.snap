// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1eykvv3"
  >
    <div
      class="chakra-stack css-i49z84"
      data-testid="holder-actions-buttons"
    >
      <a
        class="chakra-button css-vueqt9"
        data-testid="transfer-button"
        href="/security/:id/transfer"
      >
        Transfer
      </a>
      <a
        class="chakra-button css-vueqt9"
        data-testid="redeem-button"
        href="/security/:id/redeem"
      >
        Redeem
      </a>
    </div>
    <div
      class="chakra-stack css-9mll8r"
    >
      <div
        class="chakra-stack css-1flyuon"
      >
        <div
          class="chakra-stack css-3lo6qo"
          data-testid="security-details"
        >
          <div
            class="css-1dixb64"
          >
            <p
              class="chakra-text css-1an6htx"
            >
              Details
            </p>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Name"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Name
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Symbol"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Symbol
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Decimal"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Decimal
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-ISIN"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              ISIN
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-ID"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              ID
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Currency"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Currency
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Nominal Value"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Nominal Value
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Max Supply"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Max Supply
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Total Supply"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Total Supply
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Pending to be minted"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Pending to be minted
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1flyuon"
      >
        <div
          class="chakra-stack css-17irsum"
        >
          <div
            class="css-146esfn"
          >
            <p
              class="css-dmlij3"
            >
              Total Supply
            </p>
          </div>
          <div
            class="css-1jif775"
          >
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-3lo6qo"
        >
          <div
            class="css-1dixb64"
          >
            <p
              class="chakra-text css-1an6htx"
            >
              Permissions
            </p>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Controllable"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Controllable
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Blocklist"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Blocklist
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Approval list"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Approval list
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-3lo6qo"
        >
          <div
            class="css-1dixb64"
          >
            <p
              class="chakra-text css-1an6htx"
            >
              Regulations
            </p>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Regulation type"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Regulation type
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Regulation sub-type"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Regulation sub-type
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Blocked countries"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Blocked countries
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Deal size"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Deal size
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Accredited investors"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Accredited investors
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Max non-accredited investors"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Max non-accredited investors
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Manual investor verification"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Manual investor verification
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-International investors"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              International investors
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
          <div
            class="css-tvba8"
            data-testid="definition-list-item-Resale hold period"
            tabindex="0"
          >
            <p
              class="chakra-text css-vbc1qe"
            >
              Resale hold period
            </p>
            <div
              class="chakra-skeleton__group css-1rr4qq7"
            >
              <div
                class="chakra-skeleton css-5dkehc"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
